Automattic/woocommerce-payments
# Negative balance deposit notice will only show when total balance is below 0

Fixes #6941 

#### Background
If a merchant initiates an Instant Deposit, the merchant can be left in a place where available funds are below $0 and the pending funds are above $0.

![image](https://github.com/Automattic/woocommerce-payments/assets/57298/76d04681-3606-458e-a049-2f646ffb918c)

WooPayments charges a fee for instant deposits. In the case above, $2.44 is the fee which has been paid out of the available balance, leading to the negative value (which before the instant deposit was $0.) The pending balance of $2.44 is the remainder left after the Instant deposit. 

#### Changes proposed in this Pull Request

> Deposits may be interrupted while your WooPayments balance remains

A BannerNotice that shows when the merchants balance is negative only takes into account the available balance. This PR updates this logic to take into account the total (pending+available) balance.

#### Testing instructions

* Make sure instant deposits are enabled (I forced this by enabling `instant_deposits_eligible` on the server.)
* Start with a 0 total balance.
* Purchase a cart worth around $100 (to stay within the instant deposit limits.)
* Initiate an instant deposit
* Confirm that available and pending balances total to $0.
* With this change, the banner notice will not show.

![Screenshot 2024-02-02 at 2 45 22 PM](https://github.com/Automattic/woocommerce-payments/assets/57298/b90fd7de-c5f1-43fe-9949-52cb2caab2e9)
![Screenshot 2024-02-02 at 2 45 31 PM](https://github.com/Automattic/woocommerce-payments/assets/57298/8cbf1ee0-fbe4-4b43-831e-a9b0ebc429b6)

-------------------



- [x] Run `npm run changelog`

**Post merge**
